src/Stack/Build/Execute.hs

```diff
@@ -883,8 +883,9 @@ singleBench ac ee task =
             config <- asks getConfig
             cabal (console && configHideTHLoading config) ["build"]
 
+        let args = boptsAdditionalArgs (eeBuildOpts ee)
         announce "benchmarks"
-        cabal False ["bench"]
+        cabal False ["bench","--benchmark-options=" <> (intercalate " " args) <> ""]
```

Comment:
This won't properly handle arguments that have spaces in them, we'd need some kind of quoting mechanism to be used instead. I'm also unsure what the `<> ""` at the end is trying to do.